packages/gatsby-plugin-page-creator/src/derive-path.ts

```diff
@@ -32,10 +31,8 @@ export function derivePath(
   // 3.  For each slug parts get the actual value from the node data
   slugParts.forEach(slugPart => {
     // 3.a.  this transforms foo__bar into foo.bar
-    const key = compose(
-      extractFieldWithoutUnion,
-      switchToPeriodDelimiters
-    )(slugPart)
+    const cleanedField = extractFieldWithoutUnion(slugPart)[0]
```

Comment:
As `slugParts` is already an array of `{ }` segments one can always use the `[0]` from `extractFieldWithUnion`.

So `slugPart` will ever only be one `{ }` segment, not multiple, thus the extraction will only return one item.